java/code/src/com/redhat/rhn/frontend/action/CSVDownloadAction.java

```diff
@@ -129,7 +127,16 @@ protected List<SystemSearchResult> getPageData(HttpServletRequest request, HttpS
         if (null == paramPageData) {
             throw new IllegalArgumentException("Missing request parameter, " + EXPORT_COLUMNS);
         }
-        List<SystemSearchResult> pageData = (List<SystemSearchResult>) session.getAttribute(paramPageData);
+
```

Comment:
Nitpicking: this whole end block can be simplified by using `ListUtils.typedList()`, which checks the elements in the list and throws an `IllegalArgumentException` if any of them are of the wrong type. It will go through all the elements, so it might not be the best approach in case of large data sets, but it is more explicit on what you're trying to achieve:

```java
        List<BaseDto> pageData = (List<BaseDto>) session.getAttribute(paramPageData);
        if (null == pageData) {
            throw new IllegalArgumentException("Missing value for session attribute, " + paramPageData);
        }
        // Check the actual runtime type of the elements
        return ListUtils.typedList(pageData, BaseDto.class);
```

Also, if we are not sure about the type we should define `pageData` as `List<?>` until we check it, but I guess that after your change we don't expect the type to be different, right?

Reply:
Thanks for the tip! It looks cleaner now.

The decision to use BaseDto as the generic type of pageData was based on the observation that all objects we want to download as CSV are descendant of it. Using List<?> is a valid option, but I decided not to use it to have more type safety and get rid of all the IDE warnings.